### PR TITLE
Add AddGameMasterModal

### PR DESCRIPTION
**Pull request by ECWireless**
- Relies on this subgraph PR: https://github.com/raid-guild/character-sheets-subgraph/pull/14

### CODE REVIEW

File: pages/games/[gameId].tsx [hunk 1]
```diff
@@ -205,8 +214,8 @@ function GamePage(): JSX.Element {
                 >
                   {shortenAddress(id)}
                 </Link>
-                {isMaster && (
-                  <Button onClick={updateGameMetadata.onOpen} size="sm">
+                {(isAdmin || isMaster) && (
```
Comment (dan13ram) [inline pages/games/[gameId].tsx:217]: or isOwner ?

Comment (dan13ram) [inline pages/games/[gameId].tsx:217]: or isOwner ?

File: components/Modals/AddGameMasterModal.tsx [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+        if (!isAdmin) throw new Error('You are not a game admin');
+
+        setIsAdding(true);
+
+        const transactionhash = await walletClient.writeContract({
+          chain: walletClient.chain,
+          account: walletClient.account?.address as Address,
+          address: game.gameMasterHatEligibilityModule as Address,
+          abi: parseAbi([
+            'function addEligibleAddresses(address[] calldata _addresses) external',
```
Comment (dan13ram) [inline components/Modals/AddGameMasterModal.tsx:106]: Doesn't this require another transaction for minting the game master hat?

Comment (ECWireless) [reply components/Modals/AddGameMasterModal.tsx:106]: Ah, ya, forgot to mention that this assumes adding eligibility and minting a new hat will be wrapped into a single function. Maybe for now, I'll just link to the hats dashboard after the first transaction is complete

File: pages/games/[gameId].tsx (hunk 1 above)
Comment (ECWireless) [reply pages/games/[gameId].tsx]: Add in bc64c93db8a8398f5e6942b507d76b90e7a973f6

File: components/Modals/AddGameMasterModal.tsx (hunk 2 above)
Comment (ECWireless) [reply components/Modals/AddGameMasterModal.tsx:107]: Added in bc64c93db8a8398f5e6942b507d76b90e7a973f6

**Pull request merged**